meshtastic/tests/test_node_runtime_seturl.py

```
@@ -0,0 +1,1612 @@
+
+
+class TestSetUrlParser:
+    """Tests for _SetUrlParser."""
+
+    @pytest.mark.unit
+    def test_parse_with_valid_url_returns_parsed_input(self) -> None:
+        """parse() with valid URL returns parsed config."""
+        url = _make_valid_channel_set_url("testchannel")
+
+        def raise_error(msg: str) -> NoReturn:
+            raise AssertionError(f"Should not raise: {msg}")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)